src/OpenIddict/OpenIddictExtensions.cs

```diff
@@ -38,7 +38,29 @@ public static IdentityBuilder AddOpenIddict<TApplication>([NotNull] this Identit
                 // and the MVC modules are enabled.
                 builder.UseAssets();
                 builder.UseMvc();
+                configuration(builder);
+                
+                //This must be called AFTER configuration(builder) or it overwrites the user's security header configuration.
+                builder.UseSecurityHeaders();
+            });
+        }
 
+        /// <summary>
+        /// This method will enable all the default OpenIddict options EXCEPT for the NWebsec security headers
+        /// such as the Content Security Policy headers. This means you must implement all headers yourself if
+        /// you want them.
+        /// </summary>
+        /// <param name="app"></param>
+        /// <param name="configuration"></param>
+        /// <returns></returns>
+        public static IApplicationBuilder UseOpenIddictNoSecurityHeaders(
```

Comment:
I don't think we should offer such a method. If you *really* want to prevent the security headers from being applied, using `UseOpenIddictCore` to roll your own configuration is the way to go.